tutorials/plot_sleep.py

```diff
@@ -0,0 +1,115 @@
+"""
+Example on sleep data
+=====================
+
+XXX add formal description of what we do here
+
+"""
+
+# Authors: Alexandre Gramfort <alexandre.gramfort@inria.fr>
+#
+# License: BSD Style.
+
+import pandas as pd
+import matplotlib.pyplot as plt
+
+import mne
+from mne.datasets.sleep_physionet import fetch_data
+
+psg_fname, hyp_fname = fetch_data(subjects=[0])[0]
+
+raw = mne.io.read_raw_edf(psg_fname, stim_channel=False)
+annotations = mne.read_annotations(hyp_fname)
+
+##############################################################################
+# preprocessing annotations
```

Comment:
I don't understand all this pre-processing. What is its purpose?

if we only want to merge `Sleep stage 3` and `Sleep stage 4` we should be able to do so with a function. 

And I don't understand the resampling problem.

Reply:
The problem comes from the format of annotations. Sleep stages are traditionally annotated over 30s of PSG signal (by both experts and algorithms). For this reason, it could be useful to output annotations already resampled and associated to 30s of signals.

Regarding the merging of sleep stage 3 and 4, nowadays people generally work with 5 sleep stages instead of 6 and merge potential samples of label 'sleep stage 4' with samples of label 'sleep stage 3'.

A way to overcome the resampling problem might be to have a resampling parameter inside the `read_annotations` function that allows the user to get already resampled annotations. What do you think ?

Reply:
I don't really like it. `read_annotations` should only read. If this is a type of analysis is always needed then we should do some preprocessing or something. 

This example also makes me question if we want to always read the annotations of a file. Maybe this is better:

```py
psg_annotations = read_annotations(edf_file).to_psg()
raw = read_raw_edf(edf_file, annotations=psg_annotations)
```
or 
```py
psg_annotations = read_annotations(edf_file).to_psg()
raw = read_raw_edf(edf_file)
raw.set_annotations(psg_annotations)
```
instead of 

```py
raw = read_raw_edf(edf_file)
raw.annotations.to_psg(inplace=True)
```

Reply:
> I don't really like it. read_annotations should only read.

Agreed

> This example also makes me question if we want to always read the annotations of a file. 

I think we should. We should avoid adding options / kwargs to `read_raw_*` to modify how things are read / represented. The job of the `read_raw_*` functions should ideally be to transparently read all data from disk in a given file, in as close to the storage format as possible. Then we can have other functions to modify `Raw` instances as necessary for use cases. (For example, the `montage` argument of `read_raw_*` should really not be there, either -- we should do `raw.set_montage(...)` instead.)

Of your three code snippets, this one looks cleanest to me:
```
raw = read_raw_edf(edf_file)
raw.annotations.to_psg(inplace=True)
```
But I think the following is better, since it's more explicit and avoids an otherwise redundant `inplace` kwarg:
```
raw = read_raw_edf(edf_file)
raw.set_annotations(raw.annotations.to_psg())
```

Reply:
I am -1 on the to_psg method. This adds some Sleep specific code to an Annotations object that should be agnostic.

I'd rather prefer a 

df = annotations.to_dataframe()

and then you use df.resample or some other pandas code to do what you want.
Then you can recreate the annotations.
If it's a mess many some simple numpy code can do the job?